components/ai_chat/core/browser/model_service_unittest.cc

```diff
@@ -385,4 +390,172 @@ TEST_F(ModelServiceTest, GetLeoModelKeyByName_And_GetLeoModelNameByKey) {
   EXPECT_FALSE(name.has_value());
 }
 
+TEST_F(ModelServiceTest, DeleteCustomModelsByEndpoint) {
+  const GURL endpoint1 = GURL("http://example.com");
+  const GURL endpoint2 = GURL("http://other.com");
+
+  // Add multiple custom models with different endpoints
+  {
+    mojom::ModelPtr model1 = mojom::Model::New();
+    model1->display_name = "Model 1";
+    model1->options = mojom::ModelOptions::NewCustomModelOptions(
+        mojom::CustomModelOptions::New("model1", 0, 0, 0, "", endpoint1, ""));
+    GetService()->AddCustomModel(std::move(model1));
+
+    mojom::ModelPtr model2 = mojom::Model::New();
+    model2->display_name = "Model 2";
+    model2->options = mojom::ModelOptions::NewCustomModelOptions(
+        mojom::CustomModelOptions::New("model2", 0, 0, 0, "", endpoint1, ""));
+    GetService()->AddCustomModel(std::move(model2));
+
+    mojom::ModelPtr model3 = mojom::Model::New();
+    model3->display_name = "Model 3";
+    model3->options = mojom::ModelOptions::NewCustomModelOptions(
+        mojom::CustomModelOptions::New("model3", 0, 0, 0, "", endpoint2, ""));
+    GetService()->AddCustomModel(std::move(model3));
+  }
+
+  {
+    auto custom_models = GetService()->GetCustomModels();
+    EXPECT_EQ(custom_models.size(), 3u);
+  }
+
+  // Delete all models with endpoint1
+  GetService()->MaybeDeleteCustomModels(base::BindRepeating(
+      [](const GURL& target_endpoint, const base::Value::Dict& model_dict) {
+        const std::string* endpoint_str =
+            model_dict.FindString(kCustomModelItemEndpointUrlKey);
+        return endpoint_str && GURL(*endpoint_str) == target_endpoint;
+      },
+      endpoint1));
```

Comment:
We can use `base::BindLambdaForTesting` so we can directly capture `endpoint1` and make test codes cleaner. Same suggestion applies to the rest of the test file.

Reply:
fixed in 1e06e1da0d127d10925c52568ca26e08dd7fdb37